snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/exporter/Rf2Exporter.java

```diff
@@ -156,28 +217,121 @@ public final void exportBranch(
 					.execute(context);
 				});
 				
-				while (iterator.hasNext()) {
-					final R hits = iterator.next();
-					
-					getMappedStream(hits, context, branch)
-						.forEachOrdered(row -> {
-							String id = row.get(0);
-							String effectiveTime = row.get(1);
-							
-							if (!visitedComponentEffectiveTimes.add(String.format("%s_%s", id, effectiveTime))) {
-								return;
-							}
-							
-							try {
-								fileChannel.write(toByteBuffer(TAB_JOINER.join(row)));
-								fileChannel.write(toByteBuffer(CR_LF));
-							} catch (final IOException e) {
-								throw new SnowowlRuntimeException("Failed to write contents for file '" + exportFile.getFileName() + "'.");
-							}
-						});
-				}
+				exportBatch(branch, context, iterator, visitedComponentEffectiveTimes, fileChannel, exportFile.getFileName());
+			}
+		}
+	}
+	
+	private String getLatestVersionPath(RepositoryContext context, String branchPath) {
```

Comment:
Hmm, it would be better to let the request compute the necessary paths to consider when exporting and let the Rf2Exporter just export the data. Now we have a mixed implementation due to one use case, which makes the class harder to understand. We need to come up with a better solution.